### PR TITLE
Handle OPTIONS 200 level requests

### PR DESCRIPTION
**Pull request by nadiacollado**
- Handles OPTIONS 200 level requests via allow header
- Refactors ResponseWriter tests to include allow header
- Refactors isRequestValid
- Refactors formatResponse
- Adds private functions hasBody() and hasHeaders()
- Fixes typo in Constants class from constants to Constants

### CODE REVIEW

File: src/main/java/httpserver/response/ResponseWriter.java [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
-                    .addHeader("Content-Type", "text/html")
+                    .addHeader("Allow", methods)
                     .setBody("Hello world")
                     .build();
-
         } else {
             response = new ResponseBuilder()
                     .setStatusCode(StatusCodes.SUCCESS)
-                    .addHeader("Content-Type", "text/html")
+                    .addHeader("Allow", methods)
```
Comment (ranizilpelwar) [inline src/main/java/httpserver/response/ResponseWriter.java:22]: Another magic string for next time :)

File: src/main/java/httpserver/response/ResponseWriter.java [hunk 2]
```diff
@@ -34,17 +33,52 @@ public Response buildPageNotFoundResponse(Request request) {
...
-            formattedResponse = statusLine + response.body;
+        formattedResponse.append(statusLine);
+
+        if (hasHeaders(response) && (!hasBody(response))) {
+            headers = stringifyHeaders(response);
+            formattedResponse.append(headers);
+            return formattedResponse.toString();
+        }
+
+        if (hasBody(response)){
```
Comment (ranizilpelwar) [inline src/main/java/httpserver/response/ResponseWriter.java:47]: Nice refactor

File: src/main/java/httpserver/router/Router.java [hunk 3]
```diff
@@ -20,20 +20,24 @@ public Router() {
...
-                for (int i = 0; i < methodsForPath.length; i++) {
-                    if (methodsForPath[i].equals(request.method)) {
-                        return true;
-                    }
+        if (routes.containsKey(request.path)) {
+            String[] methodsForPath = routes.get(request.path);
+            for (int i = 0; i < methodsForPath.length; i++) {
+                if (methodsForPath[i].equals(request.method)) {
+                    return true;
                 }
```
Comment (ranizilpelwar) [inline src/main/java/httpserver/router/Router.java:34]: Glad you were able to get rid of the nested for loop!

Comment (ranizilpelwar) [reply src/main/java/httpserver/router/Router.java:34]: This is neat syntax to try out sometime: 
for (int element : arr) {
    if (element == toCheckValue) {
        return true;
    }
}

File: src/main/java/httpserver/utils/Constants.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package httpserver.utils;
+
+public class Constants {
+    public static String LINE_BREAK = "\r\n";
+    public static String DOUBLE_LINE_BREAK = "\r\n\r\n";
+    public static String PROTOCOL = "HTTP/1.1";
+}
```
Comment (ranizilpelwar) [inline src/main/java/httpserver/utils/Constants.java:7]: Love it!

**Review by ranizilpelwar — APPROVED** [review]
Overall, great job!

File: src/main/java/httpserver/response/ResponseWriter.java (hunk 1 above)
Comment (nadiacollado) [reply src/main/java/httpserver/response/ResponseWriter.java:22]: Ah, yes, will change in next ticket!

**Pull request merged**